compiler/damlc/lib/DA/Cli/Damlc/Packaging.hs

```diff
@@ -69,8 +69,34 @@ import SdkVersion
 --   ledger. Based on the DAML-LF we generate dummy interface files
 --   and then remap references to those dummy packages to the original DAML-LF
 --   package id.
-createProjectPackageDb :: NormalizedFilePath -> Options -> PackageSdkVersion -> [FilePath] -> [FilePath] -> IO ()
+
+calcUnitsFromDeps :: [FilePath] -> IO [UnitId]
+calcUnitsFromDeps deps = do
+    let (fpDars, fpDalfs) = partition ((== ".dar") . takeExtension) deps
+    dars <- mapM extractDar fpDars
+    let dalfsFromDars =
+            [ ( dropExtension $ takeFileName $ ZipArchive.eRelativePath e
+              , BSL.toStrict $ ZipArchive.fromEntry e
+              )
+            | e <- concatMap edMain dars
+            ]
+    dalfsFromFps <-
+        forM fpDalfs $ \fp -> do
+            bs <- BS.readFile fp
+            pure (dropExtension $ takeFileName fp, bs)
+    let allDalfs = dalfsFromDars ++ dalfsFromFps
```

Comment:
This is a slightly confusing name especially given the similarity to https://github.com/digital-asset/daml/blob/36e188cac84ddb0f4259cd077a7bb6b4fd466311/compiler/damlc/lib/DA/Cli/Damlc/Packaging.hs#L128. This does _not_ contain all dalfs, it only contains the main dalfs from each DAR. Maybe `rootDalfs` or just `mainDalfs` would be a better name.

Reply:
done